lib/ids.js

```diff
@@ -7,36 +11,114 @@ two numbers results in a different checksum digit, and thus an invalid id.
  - 1234 -> 12348
  - 1324 -> 13249, a transpose also changes the checksum id
 */
-var addCheckDigit = function(digits) {
-  var digitArray = digits.split('');
-
-  var offset = digitArray.length + 1;
-  var total = _.reduce(digitArray, function(sum, digit, index) {
-    return sum + (Number(digit) * (offset - index));
-  }, 0);
-  var result = total % 11;
+const addCheckDigit = (digits) => {
+  const digitArray = digits.split('');
+
+  const offset = digitArray.length + 1;
+  const total = _.reduce(
+    digitArray,
+    (sum, digit, index) => sum + (Number(digit) * (offset - index)),
+    0
+  );
+
+  const result = total % 11;
   digitArray.push(result === 10 ? 0 : result);
 
   return digitArray.join('');
 };
 
-module.exports = {
-    /*
-      Generates a random N digit ID. The last ID is a checksum digit
-    */
-    generate: function(length) {
-      if (length && typeof length !== 'number') {
-        throw new Error('generate requires that you pass it a length 0 < x < 14');
-      }
+const generateId = function(length) {
+  if (length && typeof length !== 'number') {
+    throw new Error('generate requires that you pass it a length 0 < x < 14');
+  }
 
-      if (length >= 14) {
-        // TODO: when we support a million billion patients per installation change
-        //       the algorithm to support it ;-)
-        throw new Error('WARNING: id length of ' + length + ' is too long');
+  if (length >= 14) {
+    // TODO: when we support a million billion patients per installation change
+    //       the algorithm to support it ;-)
+    throw new Error('WARNING: id length of ' + length + ' is too long');
+  }
+
+  const randomDigits = String(Math.random() * 10).replace('.','');
+
+  return addCheckDigit(randomDigits.substring(0, length - 1));
```

Comment:
I think it's basically there now. I feel like it should be easier but I'm not sure that it is. I guess it's something people don't do very often...